includes/class-sensei-data-cleaner.php

```diff
@@ -119,4 +134,28 @@ private static function cleanup_pages() {
 			wp_trash_post( $my_courses_page_id );
 		}
 	}
+
+	/**
+	 * Cleanup data for taxonomies.
+	 *
+	 * @access private
+	 */
+	private static function cleanup_taxonomies() {
+		foreach ( self::$taxonomies as $taxonomy ) {
+			// Ensure taxonomies are registered so we can delete their terms.
+			register_taxonomy( $taxonomy, 'post' );
```

Comment:
This feels wrong, especially given that none of the taxonomies are attached to posts, but rather to Sensei's own CPTs.

Reply:
I agree, it's a bit of a hack.

I guess the big question is whether or not there is value in using `wp_delete_term`. I'm still a bit torn on this. It does a bunch of stuff, but I don't know that there's anything critical that we would really need. I'm also not sure I entirely understand everything it's doing, which makes me a little worried _not_ to use it 😄 

If we do want to use `wp_delete_term`, then we need the taxonomies to be registered. We could register each one individually in code, and even use call the appropriate Sensei functions to register them. It would be a little more work to maintain (e.g. if we add new taxonomies in the future). And honestly, it wouldn't be necessary, from a technical perspective. In order for the deletion to work, the taxonomy can be registered on anything, including `post`, which is why the above code works fine.

🤔

Reply:
Usually my primary thought when implementing hacks is "Is it possible I'm introducing some as yet unknown potentially harmful side effect by doing this?" And of course the answer is always Yes, because you just never really know until the crap hits the fan. 😅 

Having said that, the risk of doing that here is (maybe?) fairly low. I do know that the EDD guys are really smart and are well-respected in the WordPress community, so I tend to trust their code. But I'm also good with doing whatever you think is best.